lib/sendgrid/recipient.rb

```diff
@@ -15,7 +18,6 @@ def add_substitution(key, value)
     end
 
     def add_to_smtpapi(smtpapi)
-      return if @address.nil? || @substitutions.empty?
```

Comment:
What about the case where substitutions are `nil`?  It looks like the loop falls through and everything proceeds as expected.  That's fine by me, so long as clients are confused.